zerver/lib/types.py

```diff
@@ -51,3 +51,19 @@ class LinkifierDict(TypedDict):
     pattern: str
     url_format: str
     id: int
+
+
+class SAMLIdPConfigDict(TypedDict, total=False):
+    entity_id: str
+    url: str
+    attr_user_permanent_id: str
+    attr_first_name: str
+    attr_last_name: str
+    attr_username: str
+    attr_email: str
+    attr_org_membership: str
+    display_name: str
+    display_icon: str
```

Comment:
Are these actually strings or E.g. `Optional[str]`?  I guess not for this, but while looking at this, I noticed we do have a default of `None` for the `display_icon` field for OIDC.  Do we want to change the template for `SOCIAL_AUTH_OIDC_ENABLED_IDPS` to not have `display_icon: None` and instead have that property unset?

Reply:
They're string - they may or may not be in the dict because it's not `total`. We can change oidc config dict to be consistent with this typing if we want.

Reply:
I'm not sure that's correct -- they can be in the dictionary with a value of `None`, as `prod_settings_template.py` does.